Firestore/Example/Tests/API/FIRQuerySnapshotTests.mm

```diff
@@ -94,7 +100,7 @@ - (void)testIncludeMetadataChanges {
   };
 
   std::shared_ptr<Firestore> firestore = FSTTestFirestore().wrapped;
-  FSTQuery *query = FSTTestQuery("foo");
+  core::Query query = Query("foo");
```

Comment:
Alternatively this could be `auto query = Query("foo");` or `class Query query = Query("foo");`. This particular construction is vexing because there's a conflict between the class `Query` and the helper function `Query` that creates a query by parsing the string into a resource path first. In most of the other cases (like OrderBy or Filter) the generated values are temporaries so this doesn't come up. Do you have any preference? I think I actually favor `auto` but that may be abusing the style guide.

Reply:
OK, wow. It sounds like you're inclined toward what I already have in the PR, so I'll just use that :-).